### PR TITLE
[#12641] Adding coverage to the isResponseOfFeedbackQuestionVisibleToStudent method

### PR DESCRIPTION
**Pull request by TalesRG**
Fixes [#12641]

**Outline of Solution**

This PR adds unit tests for method isResponseOfFeedbackQuestionVisibleToStudent from the FeedbackResponsesLogic class.

testResponseVisibleToStudentWhenShownToOwnTeamMembers
testResponseNotVisibleToStudentWhenReceiverTeamMembers
testResponseVisibleToStudentWhenReceiverIsGiver
testResponseVisibleToStudentWhenReceiverIsOwnTeamMember
testResponseVisibleToStudentInSameSection
testResponseVisibleToStudentWhenReceiverExcludesSelf
testResponseVisibleToStudentWhenReceiverIsStudent
testResponseVisibleToStudentWhenShownToStudents

### CODE REVIEW

**Comment by cedricongjh** [comment]
hey @TalesRG do ensure that the linting checks on github pass before we proceed to review this PR, thank you!

File: src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java [hunk 1]
```diff
@@ -1712,6 +1714,123 @@ public void testGetSessionResultsForUser_orphanResponseInDB_shouldStillHandleCor
...
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(NONE)
+                .withGiverType(NONE)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, false);
+    }
+
+    @Test
+    public void testIsResponseOfFeedbackQuestionVisibleToStudentCT8(){
```
Comment (cedricongjh) [inline src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java:1820]: let's rename the test cases to be more descriptive as well, following the convention testIsResponseOfFeedbackQuestionVisibleToStudent_<what scenario the test case is testing>_<the expected result>, an example is above  on line 1687 `testGetSessionResultsForUser_orphanResponseInDB_shouldStillHandleCorrectly`

Comment (TalesRG) [reply src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java]: corrected test case descriptions

File: src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java [hunk 2]
```diff
@@ -1712,6 +1712,116 @@ public void testGetSessionResultsForUser_orphanResponseInDB_shouldStillHandleCor
...
+        showResponseTo.add(FeedbackParticipantType.OWN_TEAM_MEMBERS);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.NONE)
+                .withGiverType(FeedbackParticipantType.TEAMS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
```
Comment (cedricongjh) [inline src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java:1823]: this test case is a positive test case for 
question.getGiverType() == FeedbackParticipantType.TEAMS
|| question.isResponseVisibleTo(FeedbackParticipantType.OWN_TEAM_MEMBERS, (lines 320 and 321 of FeedbackResponsesLogic)

given that it is an OR case, let's seperate the 2 conditions into their own test case

File: src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java [hunk 3]
```diff
@@ -1712,6 +1712,116 @@ public void testGetSessionResultsForUser_orphanResponseInDB_shouldStillHandleCor
...
+        showResponseTo.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.GIVER)
+                .withGiverType(FeedbackParticipantType.STUDENTS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
```
Comment (cedricongjh) [inline src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java:1795]: let's add a test case where isStudentRecipientType is false but question.getRecipientType().isTeam() is true, and another where isStudentRecipientType is true, but question.isResponseVisibleTo(FeedbackParticipantType.RECEIVER) is false

**Review by cedricongjh — CHANGES_REQUESTED** [review]
apologies for missing this, do take a look at the new comments!

**Comment by weiquu** [comment]
Closing due to inactivity. Feel free to re-open if you want to continue working on this PR.

**Pull request closed without merging**